### PR TITLE
login: redirect if user logged in

### PR DESCRIPTION
**Pull request by ayman-alkom**

### CODE REVIEW

File: src/Controller/Component/UserToolComponent.php [hunk 1]
```diff
@@ -365,6 +365,10 @@ public function mapAction() {
  * @return bool
  */
 	public function login($options = []) {
+		$Auth = $this->_getAuthObject();
+		if($Auth->user('id')){
```
Comment (burzum) [inline src/Controller/Component/UserToolComponent.php]: Missing space in front of the if and the {.

**Comment by burzum** [comment]
This should be implemented like all the other redirects. Check how the configuration options are used and how `handleFlashAndRedirect()` is used. This is the correct way to implement that feature. I won't accept this PR because it is not configurable as everything else is in the plugin. It will *always* redirect you to the last referer, this should be configurable.

**Comment by ayman-alkom** [comment]
I tried to implement it as the others.

**Comment by burzum** [comment]
I'll do it when I find the time. You're now using the success flash when a successful login happened for the case when the user was already logged in and *removed* the message when the user logged in successfully. This requires a new message.

File: src/Controller/Component/UserToolComponent.php [hunk 2]
```diff
@@ -365,7 +365,16 @@ public function mapAction() {
...
  */
 	public function login($options = []) {
+		$Auth = $this->_getAuthObject();
+
 		$options = Hash::merge($this->_config['login'], $options);
+		if ($options['successRedirectUrl'] === null) {
+			$options['successRedirectUrl'] = $Auth->redirectUrl();
+		}
+
+		if ($Auth->user('id')) {
```
Comment (burzum) [inline src/Controller/Component/UserToolComponent.php]: You can't rely on the user `id` being present, could be a token as well for state less systems.

**Comment by burzum** [comment]
I've fixed this in the **develop** branch of the plugin, see https://github.com/burzum/cakephp-user-tools/commit/0d3c5e5ebcfabb4a2aa4054ad45026948afe0139.

Please let me know if it works or if there are issues.

If you're curious why branches are used for development [read this](http://www.cakedc.com/git-workflow) article of my former company.

**Comment by ayman-alkom** [comment]
Unfortunately it didn't worked
in addition it doesn't redirect after the login and just return to the login page

**Comment by burzum** [comment]
Try again.

**Pull request closed without merging**